src/esw/brushed_motors/ROSHandler.cpp

```diff
@@ -7,47 +7,118 @@ void ROSHandler::init(ros::NodeHandle* rosNode) {
+
+    jointDataSA.name = SANames;
+    jointDataSA.position = std::vector<double>(SANames.size(), 0);
+    jointDataSA.velocity = std::vector<double>(SANames.size(), std::nan(""));
+    jointDataSA.effort = std::vector<double>(SANames.size(), std::nan(""));
+
+    jointDataPublisherSA = n->advertise<sensor_msgs::JointState>("sa_data", 1);
+
+    // Initialize cache
+    moveCacheSubscriber = n->subscribe<sensor_msgs::JointState>("cache_cmd", 1, moveCache);
+
+    // Initialize carousel
+    moveCarouselSubscriber = n->subscribe<mrover::Carousel>("carousel_cmd", 1, moveCarousel);
+
+    // Initialize mast gimbal
+    moveMastGimbalSubscriber = n->subscribe<mrover::MastGimbal>("mast_gimbal_cmd", 1, moveMastGimbal);
 }
 
-// REQUIRES: name is a valid name
+// REQUIRES: nothing
 // MODIFIES: nothing
 // EFFECTS: Moves the RA joints in open loop
 // and publishes angle data right after.
-void ROSHandler::moveOpenLoopRACommand(const sensor_msgs::JointState::ConstPtr& msg) {
+void ROSHandler::moveRA(const sensor_msgs::JointState::ConstPtr& msg) {
 
     for (size_t i = 0; i < RANames.size(); ++i) {
         auto controller_iter = ControllerMap::controllersByName.find(RANames[i]);
         if(controller_iter != ControllerMap::controllersByName.end()) {
-            auto name = controller_iter->first;
             auto controller = controller_iter->second;
 
             controller->moveOpenLoop((float) msg->velocity[i]);
-            jointData.position[i] = controller->getCurrentAngle();
+            jointDataRA.position[i] = controller->getCurrentAngle();
         }
     }
 
-    jointDataPublisherRA.publish(jointData);
+    jointDataPublisherRA.publish(jointDataRA);
 }
 
 // REQUIRES: nothing
 // MODIFIES: nothing
-// EFFECTS: Moves a gimbal.
-void ROSHandler::moveGimbal(const mrover::GimbalCmd::ConstPtr& msg) {
-    ControllerMap::controllersByName["mast_up_down"]->moveOpenLoop((float) msg->up_down);
-    ControllerMap::controllersByName["mast_left_right"]->moveOpenLoop((float) msg->left_right);
+// EFFECTS: Moves the SA joints in open loop
+// and publishes angle data right after.
+void ROSHandler::moveSA(const sensor_msgs::JointState::ConstPtr& msg) {
+
+    for (size_t i = 0; i < SANames.size(); ++i) {
+        auto controller_iter = ControllerMap::controllersByName.find(SANames[i]);
+        if(controller_iter != ControllerMap::controllersByName.end()) {
+            auto controller = controller_iter->second;
+
+            controller->moveOpenLoop((float) msg->velocity[i]);
+            jointDataSA.position[i] = controller->getCurrentAngle();
+        }
+    }
+
+    jointDataPublisherSA.publish(jointDataSA);
+}
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: Moves the cache in open loop
+void ROSHandler::moveCache(const sensor_msgs::JointState::ConstPtr& msg) {
+    auto controller_iter = ControllerMap::controllersByName.find("cache");
+    assert(controller_iter != ControllerMap::controllersByName.end());
```

Comment:
I don't see a point in asserting this and then checking the name. Personally I kinda think we should just log an error.